solr/core/src/java/org/apache/solr/security/Permission.java

```diff
@@ -52,18 +53,21 @@ static Permission load(Map<?, ?> m) {
     p.role = readValueAsSet(m, "role");
     if (PermissionNameProvider.Name.get(name) != null) {
       p.wellknownName = PermissionNameProvider.Name.get(name);
-      HashSet<String> disAllowed = new HashSet<>(knownKeys);
-      disAllowed.remove("role"); // these are the only
-      disAllowed.remove(NAME); // allowed keys for well-known permissions
-      disAllowed.remove("collection"); // allowed keys for well-known permissions
-      disAllowed.remove("index");
-      for (String s : disAllowed) {
+      for (String s : customPermissionAdditionalKeys) {
         if (m.containsKey(s))
           throw new SolrException(
               SolrException.ErrorCode.BAD_REQUEST,
               s + " is not a valid key for the permission : " + name);
       }
+    } else if (customPermissionAdditionalKeys.stream().noneMatch(m::containsKey)) {
+      // Custom permissions must contain one of the additional keys to be valid
```

Comment:
@risdenk Do you agree with the above assumption, i.e. that a custom permission MUST have either a "path", "method" or "params" key? Perhaps the rule could be simplified to say it must have a "path" key?

Reply:
based on https://solr.apache.org/guide/solr/latest/deployment-guide/rule-based-authorization-plugin.html#custom-permissions - pretty much all parameters are optional. `path` is not required. 

it looks like `collection` is missing as one of the options to `customPermissionAdditionalKeys`

Reply:
Yea, not sure the docs are precise here. How can you configure a custom permission without a `path`? Like the below one, it does not make sense as it does not define what endpoint to protect. You'd need a `"path": "/select"` for that:

```json
{ "name": "myColl", "role": "admin", "collection": "test" }
```

And for a custom admin request (no collection), it makes no sense to specify "method" or "params" without a "path":

```json
{ "name": "myAdmin", "role": "admin", "method": "GET" }
```

Thus I believe we must require a `path` config for custom perms.

Reply:
So I changed it to require "path" only, will see if tests pass now.

Reply:
@janhoy if we are making path required (or at least making it explicit now) the docs should be updated as well?